src/containers/FormikForm/GrepCodesField.tsx

```diff
@@ -73,15 +80,14 @@ const GrepCodesField = ({ prefixFilter }: Props) => {
           code: grepCode.code,
           title: `${grepCode.code} - ${grepCode.title.title}`,
           status: "success",
-        } as GrepCode;
+        } as const;
       });
 
-      const withFailed = codes.concat(
-        newGrepCodes
-          .filter((code) => !codes.some((c) => c?.code === code))
-          .map((code) => ({ code: code, status: "error" }) as GrepCode),
-      );
-      return withFailed;
+      const failedCodes = newGrepCodes
+        .filter((code) => !codes.some((c) => c?.code === code))
+        .map((code) => ({ code: code, status: "error" }) as const);
+
+      return [...codes, ...failedCodes];
```

Comment:
Får ikke lov av ts å concate listen når typen er definert som en union type :/

Reply:
Er ikke det bare å type både `codes` og `failedCodes` som `GrepCode[]`?

Reply:
For å snu dette på hodet: Hvilken fordel får vi egentlig av å returnere begge i samme array? Blir denne løsningen bedre av å separere de i to forskjellige felt?